packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/ModelProvider.cs

```diff
@@ -37,29 +36,17 @@ public ModelProvider(InputModelType inputModel)
         {
             _inputModel = inputModel;
             Description = inputModel.Description != null ? FormattableStringHelpers.FromString(inputModel.Description) : $"The {Name}.";
-            _declarationModifiers = TypeSignatureModifiers.Partial |
-                (inputModel.ModelAsStruct ? TypeSignatureModifiers.ReadOnly | TypeSignatureModifiers.Struct : TypeSignatureModifiers.Class);
-
-            if (inputModel.Access == "internal")
-            {
-                _declarationModifiers |= TypeSignatureModifiers.Internal;
-            }
-
-            bool isAbstract = inputModel.DiscriminatorProperty is not null && inputModel.DiscriminatorValue is null;
-            if (isAbstract)
-            {
-                _declarationModifiers |= TypeSignatureModifiers.Abstract;
-            }
 
             if (inputModel.BaseModel is not null)
             {
                 _baseTypeProvider = new(() => CodeModelPlugin.Instance.TypeFactory.CreateModel(inputModel.BaseModel));
                 DiscriminatorValueExpression = EnsureDiscriminatorValueExpression();
             }
-
-            _isStruct = inputModel.ModelAsStruct;
         }
 
+        private bool? _isStruct;
+        internal bool IsStruct => _isStruct ??= DeclarationModifiers.HasFlag(TypeSignatureModifiers.Struct);
```

Comment:
this already exits via `this.Type.IsStruct` we shouldn't duplicate this.

Reply:
removed this and replaced where this was previously used with `Type.IsStruct` which works fine.